BridgeCareApp/VuejsApp/src/services/scenario.service.ts

```diff
@@ -22,36 +22,76 @@ export default class ScenarioService {
+     * @param arrayB The second array
+     */
+    private static getMissingScenarios(arrayA: Scenario[], arrayB: Scenario[]): Scenario[]  {
+        var missingScenarios: Scenario[] = [];
+            arrayA.forEach(simulation => {
+                if (!any(propEq('simulationId', simulation.simulationId), arrayB)) {
+                    missingScenarios.push(simulation);
+                }
+            });
+            return missingScenarios;
+    }
+
+    /**
+     * Removes several scenarios from mongoDB
+     * @param scenarios Array of scenarios to remove
+     */
+    private static removeMongoScenarios(scenarios: Scenario[]): AxiosPromise {
         return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
-            axiosInstance.get('api/GetScenarios')
-                .then((responseLegacy: AxiosResponse<Scenario[]>) => {
-                    if (hasValue(responseLegacy)) {
-                        var resultant: Scenario[] = [];
-                        responseLegacy.data.forEach(simulation => {
-                            if (!any(propEq('simulationId', simulation.simulationId), scenarios)) {
-                                simulation.shared = false;
-                                simulation.status = 'N/A';
-                                resultant.push(simulation);
-                            }
-                        });
+            scenarios.forEach(simulation => {
+                nodejsAxiosInstance.delete(`api/DeleteMongoScenario/${(simulation as any)._id}`)
+                    .then((res: AxiosResponse) => {
+                        if (hasValue(res)) {
+                            return resolve(res);
+                        }
+                    })
+                    .catch((error: any) => {
+                        return resolve(error.response);
+                    });
+            });
+        });
+    }
 
-                        if (resultant.length != 0) {
-                            nodejsAxiosInstance.post('api/AddMultipleScenarios', resultant)
-                                .then((res: AxiosResponse<Scenario[]>) => {
+    /**
+     * Updates mongodb to match the scenarios from the legacy database
+     */
+    static getLegacyScenarios(): AxiosPromise {
+        return new Promise<AxiosResponse<Scenario[]>>((resolve) => {
+            axiosInstance.get('api/GetScenarios')
+                .then((responseLegacy: AxiosResponse<Scenario[]>)=>{
+                    if (hasValue(responseLegacy)){
+                    this.getMongoScenarios()
+                        .then((responseMongo: AxiosResponse<Scenario[]>)=>{
+                            if (hasValue(responseMongo)) {
+                                var scenariosToAdd = this.getMissingScenarios(responseLegacy.data, responseMongo.data);
+                                var scenariosToRemove = this.getMissingScenarios(responseMongo.data, responseLegacy.data);
+                                if (scenariosToAdd.length != 0) {
```

Comment:
This is just a personal preference of mine, but I would use '> 0' as my logical check.